includes/Generator/OrderAttribution.php

```diff
@@ -0,0 +1,309 @@
+            "Mozilla/5.0 (Linux; Android 13; SAMSUNG SM-S918B) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/21.0 Chrome/110.0.5481.154 Mobile Safari/537.36",
+        );
+
+        return $user_agents[ array_rand( $user_agents ) ];
+    }
+
+    /**
+     * Get a random tablet user agent.
+     *
+     * @return string The user agent.
+     */
+    public static function get_random_tablet_user_agent() {
+        $user_agents = array(
+            "Mozilla/5.0 (iPad; CPU OS 16_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/114.0.5735.124 Mobile/15E148 Safari/604.1",
+            "Mozilla/5.0 (Linux; Android 12; SM-X906C Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/80.0.3987.119 Mobile Safari/537.36",
+        );
+
+        return $user_agents[ array_rand( $user_agents ) ];
+    }
+
+    /**
+     * Get a random desktop user agent.
+     *
+     * @return string The user agent.
+     */
+    public static function get_random_desktop_user_agent() {
+        $user_agents = array(
+            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246",
+            "Mozilla/5.0 (X11; CrOS x86_64 8172.45.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.64 Safari/537.36",
+            "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/601.3.9 (KHTML, like Gecko) Version/9.0.2 Safari/601.3.9",
+        );
+
+        return $user_agents[ array_rand( $user_agents ) ];
+    }
+
+    /**
+     * Get a random session start time based on the order creation time.
+     *
+     * @param \DateTime $order_created_date The order creation date.
+     * @return string The session start time.
+     */
+    public static function get_random_session_start_time( $order ) {
+        $order_created_date = $order->get_date_created();
+        $session_start_hour = self::calculate_and_randomise_sesion_start_time( $order_created_date->format('H:i:s') );
+        $session_start_time = $order_created_date->format('Y-m-d') . ' ' . $session_start_hour;
+        return $session_start_time;
+    }
+
+    /**
+     * Get a random session start time based on the order creation time.
+     *
+     * @param string $endTime The order creation time.
+     * @return string The session start time.
+     */
+    public static function calculate_and_randomise_sesion_start_time( $endTime ) {
+        // Convert the end time to seconds
+        list( $hours, $minutes, $seconds) = explode( ':', $endTime );
+        $endTimeInSeconds = $hours * 3600 + $minutes * 60 + $seconds;
+
+        // Generate a random time in seconds between midnight and the end time
+        $randomTimeInSeconds = rand( 0, $endTimeInSeconds );
+
+        // Convert the random time back to HH:MM:SS format
+        $hours = floor( $randomTimeInSeconds / 3600 );
+        $minutes = floor( ( $randomTimeInSeconds / 60 ) % 60 );
```

Comment:
PHP 8.1 gives this error: `PHP Deprecated:  Implicit conversion from float … to int loses precision … on line 303`
I guess `mod`ing a float with an integer [throws that deprecated warning](https://stackoverflow.com/a/73653527).
```suggestion
        $minutes = floor( $randomTimeInSeconds / 60 ) % 60;
```

Reply:
Adding `intval` to the mix should fix this.